### PR TITLE
Support sorted column for bytes

### PR DESCRIPTION
**Pull request by xiangfu0**
Support bytes type in sorted column type.

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java [hunk 1]
```diff
@@ -600,6 +601,29 @@ public void destroy() {
     return intIterators;
   }
 
+  private IntIterator[] getSortedBitmapIntIteratorsForBytesColumn(String column) {
+    MutableDictionary dictionary = _dictionaryMap.get(column);
+    int numValues = dictionary.length();
+    IntIterator[] intIterators = new IntIterator[numValues];
+    RealtimeInvertedIndexReader invertedIndex = _invertedIndexMap.get(column);
+
+    ByteArray[] values = new ByteArray[numValues];
```
Comment (kishoreg) [inline pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java:610]: can we not sort it without creating a copy? use a comparator which will get byte[] and you can use convert it to ByteArray[].

File: pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java [hunk 2]
```diff
@@ -600,6 +601,29 @@ public void destroy() {
     return intIterators;
   }
 
+  private IntIterator[] getSortedBitmapIntIteratorsForBytesColumn(String column) {
+    MutableDictionary dictionary = _dictionaryMap.get(column);
+    int numValues = dictionary.length();
+    IntIterator[] intIterators = new IntIterator[numValues];
```
Comment (kishoreg) [inline pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java:607]: why do we need this complex object here?

**Review by kishoreg — COMMENTED** [review]
add a simple test case

File: pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java (hunk 1 above)
Comment (xiangfu0) [reply pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java:610]: This is to follow the pattern of other types. ByteArray will only take byte[] ref but not copy the content.

File: pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java (hunk 2 above)
Comment (xiangfu0) [reply pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java:607]: This is to get a docId iterator for each dictId right?

**Review by kishoreg — APPROVED** [review]
I will fix the redudant data structures across all the sorted id iterators in another PR

**Comment by kishoreg** [comment]
Ideally, the sorting should not require us to create additional arrays other than an array to store the docIds.

   ```
int[] docIds = new int[_numDocsIndexed];
    for (int i = 0; i < docIds.length; i++) {
      docIds[i] = i;
    }
    IntArrays.quickSort(docIds, new IntComparator() {
      @Override
      public int compare(int docId1, int docId2) {
        byte[] bytes1 = _dictionaryMap.get(column).getBytesValue(docId1);
        byte[] bytes2 = _dictionaryMap.get(column).getBytesValue(docId2);
        return ByteArray.compare(bytes1, bytes2);
      }

      @Override
      public int compare(Integer o1, Integer o2) {
        return 0;
      }
    });
```

**Pull request merged**